pkg/cloud/scope/machine.go

```diff
@@ -337,3 +337,10 @@ func (m *MachineScope) AWSMachineIsDeleted() bool {
 func (m *MachineScope) IsEKSManaged() bool {
 	return m.InfraCluster.InfraCluster().GetObjectKind().GroupVersionKind().Kind == "AWSManagedControlPlane"
 }
+
+// SetInterruptible sets the AWSMachine status Interruptible
+func (m *MachineScope) SetInterruptible() {
+	if m.AWSMachine.Spec.SpotMarketOptions != (&infrav1.SpotMarketOptions{}) {
```

Comment:
If `m.AWSMachine.Spec.SpotMarketOptions` is nil then interuptible will still be set which based on the comments in the issue would be wrong.